Lesson 4 HW

grrrvahrrr opened the pull request:
Задание вроде получилось, не без помощи интернета, правда. Самая большая проблема была с тем чтобы считать то, что возвращает Scanf - сначала была идея просто запросить числа, посчитать количество элементов из чего получить длину слайса и потом его создать, но так не получилось, поэтому длину вводит сам пользователь и, к сожалению если он введет больше элементов, чем длина слайса, то программа выполнится с багом - не получилось придумать как этого избежать.

Для сортировки сначала поэкспериментировал с нарезкой слайсов (закомментированный код), но он быстро превратился в череду if-ов и не всегда давал нужный результат, хотя иногда работал)
Поэтому почитав в интернете вспомнил что в Го можно менять местами значения переменных и с этим все получилось довольно коротко.

grrrvahrrr committed:
Lesson 4 HW

goarchitecture commented on lesson4/main.go:
```diff
@@ -0,0 +1,77 @@
+
+	fmt.Println(`Please, enter number of positive integers to sort (more than 1)`)
+
+	var length int
```
не понял, что именно не получилось при считывании слайса, а потом вычисления длины. Напишите конкретно какие проблемы были

goarchitecture commented on lesson4/main.go:
```diff
@@ -0,0 +1,77 @@
+
+	}
+
+	for i := 1; i < length; i++ {
```
сортировку давайте выполним в отдельной функции. мы уже знаем как их писать

goarchitecture commented on lesson4/main.go:
```diff
@@ -0,0 +1,77 @@
+
+	fmt.Println(`Please, input integers divided by "space"`)
+
+	toSort := make([]int, length)
```
очень сложный способ "считывания".
Как вариант могу предложить такой:

1) Считайте просто строку, чтобы в ней были числа через запятую (можно с пробелами)
2) затем эту строку разделяете (Split) по запятой
3) а затем в цикле каждую частичку (разделенной строки) переводите в число

grrrvahrrr replied:
Переделал ДЗ полностью в итоге. Так же кидаю ссылку на коммит.
https://github.com/grrrvahrrr/GolangLvl1/pull/4/commits/8c4757f10414cc4f89dc1c1a0f7068032f878185

grrrvahrrr replied:
Подчистил пустые строки
https://github.com/grrrvahrrr/GolangLvl1/pull/4/commits/79b63534a07410941bc6d17eb177b496392af160

goarchitecture reviewed (changes requested):
Хороший старт. Комментарии добавлены

grrrvahrrr committed:
Lesson 4 HW Update

В итоге я не придумал как сделать это используя Скан, поэтому обратился к Буфио. В итоге можно ввести числа через пробел и они будут отсортированы. В итоге, заново просто задание переписал)

grrrvahrrr committed:
Cleared a bit

Подчистил пустые строки

goarchitecture reviewed (approved):
отлично
